packages/fx-core/src/plugins/resource/localdebug/index.ts

```diff
@@ -75,122 +67,10 @@ export class LocalDebugPlugin implements Plugin {
-      if (vscEnv === VsCodeEnv.codespaceBrowser || vscEnv === VsCodeEnv.codespaceVsCode) {
-        const codespaceName = await getCodespaceName();
-
-        localTabEndpoint = getCodespaceUrl(codespaceName, 3000);
-        localTabDomain = new URL(localTabEndpoint).host;
-        localAuthEndpoint = getCodespaceUrl(codespaceName, 5000);
-        localFuncEndpoint = getCodespaceUrl(codespaceName, 7071);
-      } else {
-        localTabDomain = "localhost";
-        localTabEndpoint = "https://localhost:3000";
-        localAuthEndpoint = "http://localhost:5000";
-        localFuncEndpoint = "http://localhost:7071";
-      }
-
-      if (includeAuth) {
-        ctx.localSettings?.auth?.set(
-          LocalSettingsAuthKeys.SimpleAuthServiceEndpoint,
-          localAuthEndpoint
-        );
-      }
-
-      if (includeFrontend) {
-        ctx.localSettings?.frontend?.set(LocalSettingsFrontendKeys.TabEndpoint, localTabEndpoint);
-        ctx.localSettings?.frontend?.set(LocalSettingsFrontendKeys.TabDomain, localTabDomain);
-      }
-
-      if (includeBackend) {
-        ctx.localSettings?.backend?.set(
-          LocalSettingsBackendKeys.FunctionEndpoint,
-          localFuncEndpoint
-        );
-      }
-
-      if (includeBot) {
-        if (skipNgrok === undefined) {
-          skipNgrok = false;
-          ctx.localSettings?.bot?.set(LocalSettingsBotKeys.SkipNgrok, skipNgrok);
-        }
-
-        if (skipNgrok) {
-          const localBotEndpoint = ctx.localSettings?.bot?.get(
-            LocalSettingsBotKeys.BotEndpoint
-          ) as string;
-          if (localBotEndpoint === undefined) {
-            const error = LocalBotEndpointNotConfigured();
-            TelemetryUtils.sendErrorEvent(TelemetryEventName.localDebug, error);
-            return err(error);
-          }
-
-          const botEndpointRegex = /https:\/\/.*(:\d+)?/g;
-          if (!botEndpointRegex.test(localBotEndpoint)) {
-            const error = InvalidLocalBotEndpointFormat(localBotEndpoint);
-            TelemetryUtils.sendErrorEvent(TelemetryEventName.localDebug, error);
-            return err(error);
-          }
-
-          ctx.localSettings?.bot?.set(LocalSettingsBotKeys.BotEndpoint, localBotEndpoint);
-          ctx.localSettings?.bot?.set(LocalSettingsBotKeys.BotDomain, localBotEndpoint.slice(8));
-        } else {
-          const ngrokHttpUrl = await getNgrokHttpUrl(3978);
-          if (!ngrokHttpUrl) {
-            const error = NgrokTunnelNotConnected();
-            TelemetryUtils.sendErrorEvent(TelemetryEventName.localDebug, error);
-            return err(error);
-          } else {
-            ctx.localSettings?.bot?.set(LocalSettingsBotKeys.BotEndpoint, ngrokHttpUrl);
-            ctx.localSettings?.bot?.set(LocalSettingsBotKeys.BotDomain, ngrokHttpUrl.slice(8));
-          }
-        }
-      }
-    }
-
-    TelemetryUtils.sendSuccessEvent(TelemetryEventName.localDebug, telemetryProperties);
-    return ok(undefined);
+    return ok(Void);
```

Comment:
Curious - what's the difference between `ok(undefined)` and `ok(Void)`? Any behavior change?

Reply:
Here, they are no difference, I just want to follow the API v2 design.